src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -136,63 +159,172 @@ private int getNextFileLength() {
         return bytes;
     }
 
-    public Simulator(int opCount, Random  random, Supplier<Simulation> getNextSimulation, long meanFileLength, FileSystem referenceFileSystem, FileSystem testFileSystem) {
-        if (! referenceFileSystem.user().equals(testFileSystem.user()))
-            throw new IllegalStateException("Users must be the same");
 
-        simulatedDirectoryToFiles.put(Paths.get("/"+ referenceFileSystem.user()), new ArrayList<>());
+    private static class FileSystems {
+        private final List<Pair<FileSystem, FileSystem>> userFileSystems;
```

Comment:
do the pairs signify friends?

Reply:
got it, they are the peergos and native version of the same user. Is there a var name that would make this clear?

Reply:
done